backends/tfhe-cuda-backend/cuda/include/zk/zk_utilities.h

```diff
@@ -0,0 +1,217 @@
+#ifndef ZK_UTILITIES_H
+#define ZK_UTILITIES_H
+
+#include "../integer/integer_utilities.h"
+#include "integer/integer.cuh"
+#include <cstdint>
+
+template <typename Torus> struct zk_expand {
+  int_radix_params params;
+  uint32_t num_lwes;
+  uint32_t num_compact_lists;
+
+  int_radix_lut<Torus> *message_and_carry_extract_luts;
+
+  Torus *tmp_expanded_lwes;
+  Torus *tmp_ksed_small_to_big_expanded_lwes;
+
+  uint32_t *h_compact_list_length_per_thread;
+  uint32_t *d_compact_list_length_per_thread;
+
+  uint32_t *d_lwe_compact_input_indexes;
+
+  zk_expand(cudaStream_t const *streams, uint32_t const *gpu_indexes,
+            uint32_t gpu_count, int_radix_params params,
+            const uint32_t *num_samples_per_compact_list,
```

Comment:
What does this num_samples_per_compact_list correspond to? Is it num_lwes_per_compact_list? Maybe that would be a better name?

Reply:
Yes, it is the number of LWEs inside a particular compact list. `num_lwes_per_compact_list` fits better.